### ISSUE TITLE
ENH: Use a neutral format to have lossless interface with scipp, Astropy, Xarray

### ISSUE THREAD

**Issue by loco-philippe** [pandas-dev/pandas#58217]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Each tool has a specific structure for processing multidimensional data with the following consequences:

- interfaces dedicated to each tool,
- partially processed data,
- no unified representation of data structures

### Feature Description

The proposed format  (see [jupyter notebook](https://nbviewer.org/github/loco-philippe/ntv-numpy/blob/main/example/example_ntv_numpy.ipynb), [github repository](https://github.com/loco-philippe/ntv-numpy/blob/main/README.md), [PyPI package](https://pypi.org/project/ntv-numpy/) ) is based on the following principles:

- neutral format available for tabular or multidimensional tools (e.g. Numpy, pandas, xarray, scipp, astropy),
- taking into account a wide variety of data types as defined in [NTV](https://www.ietf.org/archive/id/draft-thomy-json-ntv-02.html) format,
- high interoperability: reversible (lossless round-trip) interface with tabular or multidimensional tools,
- reversible and compact JSON format,
- Ease of sharing and exchanging multidimensional and tabular data,

### Alternative Solutions

A first tool exists ([NTV-pandas](https://github.com/loco-philippe/ntv-pandas/blob/main/README.md)) but only deals with the JSON interface and the analysis of tabular structures. It would therefore be extended to multidimensional structures.

### Additional Context

https://github.com/numpy/numpy/issues/12481#issuecomment-2049179803
https://github.com/astropy/astropy/issues/16286
https://github.com/pydata/xarray/issues/8927
https://github.com/scipp/scipp/issues/3422

**Comment by jbrockmendel** [comment]
Can you expand on what you're asking for here?  I'm guessing you want a pd.read_this_new_thing and DataFrame.to_this_thing?

**Comment by loco-philippe** [comment]
Currently, there is a first level of support:
- in Pandas: reference in the ['NTV-pandas'](https://pandas.pydata.org/community/ecosystem.html#data-io-for-pandas) ecosystem (custom accessors)
- in Xarray: integration in the Xarray [user guide](https://docs.xarray.dev/en/latest/user-guide/pandas.html#lossless-and-reversible-conversion)

The desired integration can indeed include the IO functions: pd.read_this_new_thing and DataFrame.to_this_thing, as well as a DataFrame analysis function to identify the multidimensional structures underlying the DataFrame.

The level of integration via custom accessors may be sufficient unless you specifically want to integrate this structure analysis function (added value for Pandas) or if you want to integrate an export function into Pandas for multidimensional tools like Xarray (which does not currently exist).

**Comment by jbrockmendel** [comment]
> or if you want to integrate an export function into Pandas for multidimensional tools like Xarray (which does not currently exist).

DataFrame.to_array exists.  Do you mean something different?

**Comment by loco-philippe** [comment]
Yes, DataFrame.to_array exists but as explain in the Xarray [user guide - Lossless and reversible conversion](https://docs.xarray.dev/en/latest/user-guide/pandas.html#lossless-and-reversible-conversion), a non-dimension `coordinate` is converted into `variable` because DataFrame.to_array does not parse the tabular structure (eg it does not identify a non-dimension `coordinate` and simply converts it to a `variable`).
